batchupload/uploader.py

```diff
@@ -172,36 +184,173 @@ def up_all(in_path, cutoff=None, target='Uploaded', file_exts=None,
         if verbose:
             pywikibot.output(result.get('log'))
 
-        flog.write('%s\n' % result.get('log'))
+        flog.write(result.get('log'))
         os.rename(f, os.path.join(target_dir, base_name))
         os.rename(info_file, os.path.join(target_dir, base_info_name))
         counter += 1
-        flog.flush()
 
-    flog.close()
-    pywikibot.output('Created %s' % logfile)
+    pywikibot.output(flog.close_and_confirm())
+
+
+def up_all_from_url(info_path, cutoff=None, target='upload_logs',
+                    file_exts=None, verbose=False, test=False,
+                    target_site=None):
+    """
+    Upload all images provide as urls in a make_info json file.
+
+    Media (image) files and metadata files with the expected extension .info
+    should be in the same directory. Metadata files should contain the entirety
+    of the desired description page (in wikitext).
+
+    Outputs separate logfiles for files triggering errors, warnings (and
+    successful) so that these can be used in latter runs.
+
+    @param info_path: path to the make_info json file
+    @param cutoff: number of files to upload (defaults to all)
+    @param target: sub-directory for log files (defaults to "upload_logs")
+    @param file_exts: tuple of allowed file extensions (defaults to FILE_EXTS)
+    @param verbose: whether to output confirmation after each upload
+    @param test: set to True to test but not upload (deprecated?)
```

Comment:
I think this is leftover from a previous restructuring. Removing from here and from `up_all` docstring.